File: s07_Database/SQLiteProject/app/src/main/java/com/omeryildizce/sqliteproject/MainActivity.java
package com.omeryildizce.sqliteproject;

import androidx.appcompat.app.AppCompatActivity;

import android.database.Cursor;
import android.database.sqlite.SQLiteDatabase;
import android.os.Bundle;
import android.widget.Toast;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        try {
            SQLiteDatabase database = this.openOrCreateDatabase("Musicians", MODE_PRIVATE, null);
            database.execSQL("create table if not exists musicians (id integer primary key, name varchar, age int)");
            //database.execSQL("insert into musicians (name, age) values('Kadir', 35)");
            //database.execSQL("insert into musicians (name, age) values('Ömer', 25)");
            //database.execSQL("insert into musicians (name, age) values('Ali', 19)");
            //database.execSQL("insert into musicians (name, age) values('Ayşe', 32)");
            //database.execSQL("update musicians set name = 'Fatih' where id = 2 ");
            // database.execSQL("delete from musicians where id = 2 ");
            //Cursor cursor = database.rawQuery("select * from musicians where id = 2", null);
            Cursor cursor = database.rawQuery("select * from musicians", null);
            int nameIndex = cursor.getColumnIndex("name");
            int ageIndex = cursor.getColumnIndex("age");
            int idIndex = cursor.getColumnIndex("id");
            while (cursor.moveToNext()) {
                System.out.println("Id:  " + cursor.getInt(idIndex));
                System.out.println("Name: " + cursor.getString(nameIndex));
                System.out.println("Age:  " + cursor.getInt(ageIndex));
            }
            cursor.close();
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

File: s20_Database/SQLite/app/src/main/java/com/omeryildizce/sqlite/MainActivity.kt
package com.omeryildizce.sqlite

import android.annotation.SuppressLint
import android.database.sqlite.SQLiteDatabase
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import com.omeryildizce.sqlite.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {
    private lateinit var binding: ActivityMainBinding
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityMainBinding.inflate(layoutInflater)
        val view = binding.root
        setContentView(view)

        sqLite()
    }

    @SuppressLint("Recycle")
    private fun sqLite() {
        try {
            val myDatabase = this.openOrCreateDatabase("musicians", MODE_PRIVATE, null)
            myDatabase.execSQL("create table if not exists musicians(id INTEGER primary key,name varchar, age int)")
            // insert(myDatabase)
             delete(myDatabase)
             // update(myDatabase)

            var cursor = myDatabase.rawQuery("select * from musicians ", null)
            cursor = myDatabase.rawQuery("select * from musicians where name = 'James'", null)
            cursor = myDatabase.rawQuery("select * from musicians where id = 3", null)
            cursor = myDatabase.rawQuery("select * from musicians where name like '%ö%'", null)
            val nameIndex = cursor.getColumnIndex("name")
            val ageIndex = cursor.getColumnIndex("age")
            val idIndex = cursor.getColumnIndex("id")

            while (cursor.moveToNext()){
                val info = String.format("Id: %3s, Name: %10s, Age:%3s",cursor.getString(idIndex),cursor.getString(nameIndex),cursor.getString(ageIndex) )
                println(info)
            }


        } catch (e: Exception) {
            e.printStackTrace()
        }
    }

    private fun delete(myDatabase: SQLiteDatabase) {
        myDatabase.execSQL("delete from musicians where name = 'M Ömer'")
    }
    private fun update(myDatabase: SQLiteDatabase) {
        myDatabase.execSQL("update musicians set age = 61 where name = 'James'")
        myDatabase.execSQL("update musicians set name = 'M Ömer' where name = 'Ömer'")
    }

    private fun insert(myDatabase: SQLiteDatabase) {
            myDatabase.execSQL("insert into musicians (name, age) values ('James', 45)")
            myDatabase.execSQL("insert into musicians (name, age) values ('Ömer', 25)")
            myDatabase.execSQL("insert into musicians (name, age) values ('Ali', 35)")
            myDatabase.execSQL("insert into musicians (name, age) values ('Ayşe', 35)")
    }
}